lib/linguist/heuristics.yml

```
@@ -107,7 +107,7 @@ disambiguations:
 - extensions: ['.cls']
   rules:
   - language: TeX
-    pattern: '\\\w+{'
+    pattern: '\\ProvidesClass{'
```

On second thought, it might be wiser to limit this to start-of-line matches only. Just to be on the safe side.

```suggestion
    pattern: '^\s*\\ProvidesClass{'
```